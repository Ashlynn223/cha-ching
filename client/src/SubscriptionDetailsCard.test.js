import { render, screen } from "@testing-library/react";
import SubscriptionDetailsCard from "./SubscriptionDetailsCard";

describe("SubscriptionDetailsCard", () => {
    describe("highlight styling renders correctly", () => {
        it("does not display highlight given 'id' and 'selectedCardId' props are not passed", () => {
            render(<SubscriptionDetailsCard/>)
            expect(screen.getByRole("subscription-card")).not.toHaveClass("selected");
        })
        it("displays highlight when selected is true", () => {
            render(<SubscriptionDetailsCard id={0} selectedCardId={0} />)
            expect(screen.getByRole("subscription-card")).toHaveClass("selected");
        })
        it("does not display hightlight if card is not selected", () => {
            render(<SubscriptionDetailsCard id={1} selectedCardId={0} />)
            expect(screen.getByRole("subscription-card")).not.toHaveClass("selected");
        })
    })
    describe("displays card content correctly", () => {
        it("renders correctly", () => {
            const component = render(
                <SubscriptionDetailsCard
                    subscriptionName="starter"
                    subscriptionDetails={["daily flower delivery", "scissors", "kenzan"]}
                    price="200"
                    interval="monthly"
                />
            )
            expect(component.asFragment()).toMatchSnapshot()
        })
        it("displays subscription header", () => {
            render(<SubscriptionDetailsCard subscriptionName="starter"/>)
            expect(screen.getByText("Starter")).toBeVisible();
        })
        it("displays bulleted list", () => {
            render(<SubscriptionDetailsCard subscriptionDetails={["daily flower delivery", "scissors", "kenzan"]} />)
            expect(screen.getByText("daily flower delivery")).toBeVisible();
            expect(screen.getByText("scissors")).toBeVisible();
            expect(screen.getByText("kenzan")).toBeVisible();
        })
        it("displays price and interval of subscription", () => {
            render(<SubscriptionDetailsCard price="200" interval="monthly" />)
            expect(screen.getByText("$200 / monthly")).toBeVisible();
        })
    });
});
