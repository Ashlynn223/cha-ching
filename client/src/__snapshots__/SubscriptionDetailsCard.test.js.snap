// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubscriptionDetailsCard displays card content correctly renders correctly 1`] = `
<DocumentFragment>
  <div
    class=" container"
    role="subscription-card"
  >
    <div
      class="subscriptionName"
    >
      Starter
    </div>
    <div
      class="subscriptionPricingAndInterval"
    >
      $200 / monthly
    </div>
    <div
      class="subscriptionDetailsBulletedList"
    >
      <ul
        class="bulletedListItem"
      >
        <div
          class="bulletPointIcon"
        >
          <svg
            fill="none"
            height="23"
            viewBox="0 0 22 23"
            width="22"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              clip-path="url(#clip0_256_1504)"
            >
              <g
                filter="url(#filter0_d_256_1504)"
              >
                <path
                  d="M4.50781 11.7306C4.50781 15.331 7.42658 18.2498 11.0271 18.2498C14.6275 18.2498 17.5463 15.331 17.5463 11.7306C17.5463 8.13007 14.6275 5.2113 11.0271 5.2113C7.42658 5.2113 4.50781 8.13007 4.50781 11.7306Z"
                  fill="url(#paint0_radial_256_1504)"
                  shape-rendering="crispEdges"
                />
                <path
                  d="M4.50781 11.7306C4.50781 15.331 7.42658 18.2498 11.0271 18.2498C14.6275 18.2498 17.5463 15.331 17.5463 11.7306C17.5463 8.13007 14.6275 5.2113 11.0271 5.2113C7.42658 5.2113 4.50781 8.13007 4.50781 11.7306Z"
                  shape-rendering="crispEdges"
                  stroke="#9D6BFF"
                  stroke-width="1.34881"
                />
              </g>
              <g
                filter="url(#filter1_d_256_1504)"
              >
                <path
                  d="M8.78125 12.1801L10.1301 13.5289L13.2773 10.3817"
                  shape-rendering="crispEdges"
                  stroke="#9D6BFF"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.34881"
                />
              </g>
            </g>
            <defs>
              <filter
                color-interpolation-filters="sRGB"
                filterUnits="userSpaceOnUse"
                height="21.581"
                id="filter0_d_256_1504"
                width="21.5765"
                x="0.239109"
                y="2.73845"
              >
                <feflood
                  flood-opacity="0"
                  result="BackgroundImageFix"
                />
                <fecolormatrix
                  in="SourceAlpha"
                  result="hardAlpha"
                  type="matrix"
                  values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
                />
                <feoffset
                  dy="1.79841"
                />
                <fegaussianblur
                  stdDeviation="1.79841"
                />
                <fecomposite
                  in2="hardAlpha"
                  operator="out"
                />
                <fecolormatrix
                  type="matrix"
                  values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.25 0"
                />
                <feblend
                  in2="BackgroundImageFix"
                  mode="normal"
                  result="effect1_dropShadow_256_1504"
                />
                <feblend
                  in="SourceGraphic"
                  in2="effect1_dropShadow_256_1504"
                  mode="normal"
                  result="shape"
                />
              </filter>
              <filter
                color-interpolation-filters="sRGB"
                filterUnits="userSpaceOnUse"
                height="11.6898"
                id="filter1_d_256_1504"
                width="13.0374"
                x="4.51255"
                y="7.90886"
              >
                <feflood
                  flood-opacity="0"
                  result="BackgroundImageFix"
                />
                <fecolormatrix
                  in="SourceAlpha"
                  result="hardAlpha"
                  type="matrix"
                  values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
                />
                <feoffset
                  dy="1.79841"
                />
                <fegaussianblur
                  stdDeviation="1.79841"
                />
                <fecomposite
                  in2="hardAlpha"
                  operator="out"
                />
                <fecolormatrix
                  type="matrix"
                  values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.25 0"
                />
                <feblend
                  in2="BackgroundImageFix"
                  mode="normal"
                  result="effect1_dropShadow_256_1504"
                />
                <feblend
                  in="SourceGraphic"
                  in2="effect1_dropShadow_256_1504"
                  mode="normal"
                  result="shape"
                />
              </filter>
              <radialgradient
                cx="0"
                cy="0"
                gradientTransform="translate(11.0271 11.7306) rotate(90) scale(6.51925)"
                gradientUnits="userSpaceOnUse"
                id="paint0_radial_256_1504"
                r="1"
              >
                <stop
                  stop-color="white"
                />
                <stop
                  offset="1"
                  stop-color="white"
                  stop-opacity="0"
                />
              </radialgradient>
              <clippath
                id="clip0_256_1504"
              >
                <rect
                  fill="white"
                  height="21.581"
                  transform="translate(0.234375 0.940186)"
                  width="21.581"
                />
              </clippath>
            </defs>
          </svg>
        </div>
        <li
          class="bulletPointText"
        >
          daily flower delivery
        </li>
      </ul>
      <ul
        class="bulletedListItem"
      >
        <div
          class="bulletPointIcon"
        >
          <svg
            fill="none"
            height="23"
            viewBox="0 0 22 23"
            width="22"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              clip-path="url(#clip0_256_1504)"
            >
              <g
                filter="url(#filter0_d_256_1504)"
              >
                <path
                  d="M4.50781 11.7306C4.50781 15.331 7.42658 18.2498 11.0271 18.2498C14.6275 18.2498 17.5463 15.331 17.5463 11.7306C17.5463 8.13007 14.6275 5.2113 11.0271 5.2113C7.42658 5.2113 4.50781 8.13007 4.50781 11.7306Z"
                  fill="url(#paint0_radial_256_1504)"
                  shape-rendering="crispEdges"
                />
                <path
                  d="M4.50781 11.7306C4.50781 15.331 7.42658 18.2498 11.0271 18.2498C14.6275 18.2498 17.5463 15.331 17.5463 11.7306C17.5463 8.13007 14.6275 5.2113 11.0271 5.2113C7.42658 5.2113 4.50781 8.13007 4.50781 11.7306Z"
                  shape-rendering="crispEdges"
                  stroke="#9D6BFF"
                  stroke-width="1.34881"
                />
              </g>
              <g
                filter="url(#filter1_d_256_1504)"
              >
                <path
                  d="M8.78125 12.1801L10.1301 13.5289L13.2773 10.3817"
                  shape-rendering="crispEdges"
                  stroke="#9D6BFF"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.34881"
                />
              </g>
            </g>
            <defs>
              <filter
                color-interpolation-filters="sRGB"
                filterUnits="userSpaceOnUse"
                height="21.581"
                id="filter0_d_256_1504"
                width="21.5765"
                x="0.239109"
                y="2.73845"
              >
                <feflood
                  flood-opacity="0"
                  result="BackgroundImageFix"
                />
                <fecolormatrix
                  in="SourceAlpha"
                  result="hardAlpha"
                  type="matrix"
                  values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
                />
                <feoffset
                  dy="1.79841"
                />
                <fegaussianblur
                  stdDeviation="1.79841"
                />
                <fecomposite
                  in2="hardAlpha"
                  operator="out"
                />
                <fecolormatrix
                  type="matrix"
                  values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.25 0"
                />
                <feblend
                  in2="BackgroundImageFix"
                  mode="normal"
                  result="effect1_dropShadow_256_1504"
                />
                <feblend
                  in="SourceGraphic"
                  in2="effect1_dropShadow_256_1504"
                  mode="normal"
                  result="shape"
                />
              </filter>
              <filter
                color-interpolation-filters="sRGB"
                filterUnits="userSpaceOnUse"
                height="11.6898"
                id="filter1_d_256_1504"
                width="13.0374"
                x="4.51255"
                y="7.90886"
              >
                <feflood
                  flood-opacity="0"
                  result="BackgroundImageFix"
                />
                <fecolormatrix
                  in="SourceAlpha"
                  result="hardAlpha"
                  type="matrix"
                  values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
                />
                <feoffset
                  dy="1.79841"
                />
                <fegaussianblur
                  stdDeviation="1.79841"
                />
                <fecomposite
                  in2="hardAlpha"
                  operator="out"
                />
                <fecolormatrix
                  type="matrix"
                  values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.25 0"
                />
                <feblend
                  in2="BackgroundImageFix"
                  mode="normal"
                  result="effect1_dropShadow_256_1504"
                />
                <feblend
                  in="SourceGraphic"
                  in2="effect1_dropShadow_256_1504"
                  mode="normal"
                  result="shape"
                />
              </filter>
              <radialgradient
                cx="0"
                cy="0"
                gradientTransform="translate(11.0271 11.7306) rotate(90) scale(6.51925)"
                gradientUnits="userSpaceOnUse"
                id="paint0_radial_256_1504"
                r="1"
              >
                <stop
                  stop-color="white"
                />
                <stop
                  offset="1"
                  stop-color="white"
                  stop-opacity="0"
                />
              </radialgradient>
              <clippath
                id="clip0_256_1504"
              >
                <rect
                  fill="white"
                  height="21.581"
                  transform="translate(0.234375 0.940186)"
                  width="21.581"
                />
              </clippath>
            </defs>
          </svg>
        </div>
        <li
          class="bulletPointText"
        >
          scissors
        </li>
      </ul>
      <ul
        class="bulletedListItem"
      >
        <div
          class="bulletPointIcon"
        >
          <svg
            fill="none"
            height="23"
            viewBox="0 0 22 23"
            width="22"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              clip-path="url(#clip0_256_1504)"
            >
              <g
                filter="url(#filter0_d_256_1504)"
              >
                <path
                  d="M4.50781 11.7306C4.50781 15.331 7.42658 18.2498 11.0271 18.2498C14.6275 18.2498 17.5463 15.331 17.5463 11.7306C17.5463 8.13007 14.6275 5.2113 11.0271 5.2113C7.42658 5.2113 4.50781 8.13007 4.50781 11.7306Z"
                  fill="url(#paint0_radial_256_1504)"
                  shape-rendering="crispEdges"
                />
                <path
                  d="M4.50781 11.7306C4.50781 15.331 7.42658 18.2498 11.0271 18.2498C14.6275 18.2498 17.5463 15.331 17.5463 11.7306C17.5463 8.13007 14.6275 5.2113 11.0271 5.2113C7.42658 5.2113 4.50781 8.13007 4.50781 11.7306Z"
                  shape-rendering="crispEdges"
                  stroke="#9D6BFF"
                  stroke-width="1.34881"
                />
              </g>
              <g
                filter="url(#filter1_d_256_1504)"
              >
                <path
                  d="M8.78125 12.1801L10.1301 13.5289L13.2773 10.3817"
                  shape-rendering="crispEdges"
                  stroke="#9D6BFF"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.34881"
                />
              </g>
            </g>
            <defs>
              <filter
                color-interpolation-filters="sRGB"
                filterUnits="userSpaceOnUse"
                height="21.581"
                id="filter0_d_256_1504"
                width="21.5765"
                x="0.239109"
                y="2.73845"
              >
                <feflood
                  flood-opacity="0"
                  result="BackgroundImageFix"
                />
                <fecolormatrix
                  in="SourceAlpha"
                  result="hardAlpha"
                  type="matrix"
                  values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
                />
                <feoffset
                  dy="1.79841"
                />
                <fegaussianblur
                  stdDeviation="1.79841"
                />
                <fecomposite
                  in2="hardAlpha"
                  operator="out"
                />
                <fecolormatrix
                  type="matrix"
                  values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.25 0"
                />
                <feblend
                  in2="BackgroundImageFix"
                  mode="normal"
                  result="effect1_dropShadow_256_1504"
                />
                <feblend
                  in="SourceGraphic"
                  in2="effect1_dropShadow_256_1504"
                  mode="normal"
                  result="shape"
                />
              </filter>
              <filter
                color-interpolation-filters="sRGB"
                filterUnits="userSpaceOnUse"
                height="11.6898"
                id="filter1_d_256_1504"
                width="13.0374"
                x="4.51255"
                y="7.90886"
              >
                <feflood
                  flood-opacity="0"
                  result="BackgroundImageFix"
                />
                <fecolormatrix
                  in="SourceAlpha"
                  result="hardAlpha"
                  type="matrix"
                  values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
                />
                <feoffset
                  dy="1.79841"
                />
                <fegaussianblur
                  stdDeviation="1.79841"
                />
                <fecomposite
                  in2="hardAlpha"
                  operator="out"
                />
                <fecolormatrix
                  type="matrix"
                  values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.25 0"
                />
                <feblend
                  in2="BackgroundImageFix"
                  mode="normal"
                  result="effect1_dropShadow_256_1504"
                />
                <feblend
                  in="SourceGraphic"
                  in2="effect1_dropShadow_256_1504"
                  mode="normal"
                  result="shape"
                />
              </filter>
              <radialgradient
                cx="0"
                cy="0"
                gradientTransform="translate(11.0271 11.7306) rotate(90) scale(6.51925)"
                gradientUnits="userSpaceOnUse"
                id="paint0_radial_256_1504"
                r="1"
              >
                <stop
                  stop-color="white"
                />
                <stop
                  offset="1"
                  stop-color="white"
                  stop-opacity="0"
                />
              </radialgradient>
              <clippath
                id="clip0_256_1504"
              >
                <rect
                  fill="white"
                  height="21.581"
                  transform="translate(0.234375 0.940186)"
                  width="21.581"
                />
              </clippath>
            </defs>
          </svg>
        </div>
        <li
          class="bulletPointText"
        >
          kenzan
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
